src/main/scala/scalaz/distributed/Protocol.scala

```diff
@@ -0,0 +1,5 @@
+package scalaz.distributed
+
+trait Protocol[Type[_], Path[_, _]] {
```

Comment:
If you push this back in the module, you can get rid of the type parameters, but it's a personal preference. It may just be clunky to use since the user will never know the actual type of `Type` or `Path`.

Reply:
Moved both `Protocol` and `Metadata` back to `Client`.